synapse/storage/databases/main/relations.py

```diff
@@ -132,6 +132,44 @@ def _get_recent_references_for_event_txn(
             "get_recent_references_for_event", _get_recent_references_for_event_txn
         )
 
+    async def event_has_relations(self, event_id: str) -> bool:
+        """Check if a given event has relations.
+
+        Args:
+            event_id: The event to check.
+
+        Returns:
+            True if the event has any relations.
+        """
+
+        result = await self.db_pool.simple_select_one_onecol(
+            table="event_relations",
+            keyvalues={"event_id": event_id},
+            retcol="event_id",
+            allow_none=True,
+            desc="event_has_relations",
+        )
+        return result is not None
+
+    async def event_related_to(self, parent_id: str) -> bool:
+        """Check if a given event has other event relating to it.
+
+        Args:
+            parent_id: The event to check.
+
+        Returns:
+            True if the event is the target of any relations.
+        """
```

Comment:
I'm finding this name hard to read; from the method name I'd expect this to be a binary operator (is _ related to _? Well, literally called a relation (mathematically)...).
It doesn't seem like that, though.

Maybe this is some personal misunderstanding about what 'relations' are in Matrix; though in any case this makes it hard to approach for noobs like me.
Reckon you could extend the comment to describe what the shape of the event ID referred to by `parent_id` is meant to be; or in other words: what are 'you' (well, this function) looking for?

The same for `event_has_relations`, if you wouldn't mind, please.
It may be that a different name would fit them better, but I shouldn't be telling you that until I know what they actually mean :).

Reply:
I find the terminology very ambiguous, which makes it hard to write clear names. I'll try to make them a bit more straightforward though.

If you have a two events: `A` and `B` and `B` includes a relation to `A`, then:

* `event_related_to(A)` would return `True` (since `B` references `A`)
* `event_related_to(B)` would return `False` (since no one references `B`)
* `event_has_relations(A)` would return `False` (since `A` doesn't reference anything)
* `event_has_relations(B)` would return `True` (since `B` references `A`)

Hopefully those examples clear it up for your own understanding, but I'll try to reword and expand the docstrings.

Reply:
I pushed 558175b3702e17bbb85673560ae5215a253dc97b to attempt to clarify this, let me know if it makes sense!